### ISSUE TITLE
Inserts into bucket_bandwidth_rollups can fail because of too many retries

### ISSUE THREAD

**Issue by mniewrzal** [storj/storj#5332]
We noticed that on EU1 inserts into `bucket_bandwidth_rollups` are very often retried. Most probably its caused by large number of concurrent inserts and incorrectly design primary key (`bucket_name project_id interval_start action`). It's possible that some of inserts are dropped because of too many retries and bandwidth used by users is lower than real usage. So far we are pretty sure we are loosing just some pennies but we need to resolve that issue with that become larger issue. 
![image](https://user-images.githubusercontent.com/7313031/203074019-261916a5-4893-4c51-a768-6104c40db590.png)

Acceptance Criteria:
* We should optimize inserts into `bucket_bandwidth_rollups` to avoid deadlocks

**Comment by shaupt131** [comment]
Path forward is not yet clear.  Requires research to determine the best method to solve this issue.

**Comment by Fadila82** [comment]
The options we have:
 - Use `REGIONAL BY ROW` :  each row created has its leasholder in the region it has been created in and add a `pod_id` column. This could potentially multiply the number of rows by the number of pods (worst case scenario - bucket accessed through all satellite pods)
 - use (interval_start, project_id, bucket_name, action) as primary key - or maybe another combination of this.

**Comment by mniewrzal** [comment]
I think fastest first attempt to make things better will be altering primary key e.g. (interval_start, project_id, bucket_name, action). In theory it could be just migration change. Later we can adjust queries and/or add indexes.

I have also theory why it is hitting DB so hard. For actions like below we are sending empty bucket as a part of order. Audit, repair, GE are using segments loop where we don't know to which bucket segment belongs.
	PieceAction_GET_AUDIT
	PieceAction_GET_REPAIR
	PieceAction_PUT_REPAIR
	PieceAction_PUT_GRACEFUL_EXIT
If primary key have now `bucket` as a first column and we are sending lots of inserts with `empty bucket` it may be problematic to split traffic.

**Comment by mniewrzal** [comment]
I have some conclusions:
- hot range for EU1 is showing that problem most probably is related to inserting into empty bucket name and project id
- doing lots of inserts (from multiple pods) into the same range triggers lots of retries  
- we are doing such inserts for orders from repair, audit an GE (to empty bucket name and project id)
- its affecting each transaction with inserts because we are inserting 1000 entries at once and we have mixed actions there e.g. GET, GET_REPAIR, GET_AUDIT

What to do:
- stop inserting into bucket_bandwidth_rollups table actions different than GET and PUT
- only user actions GET and PUT are important from bucket bandwidth usage perspective, other actions are not really used anywhere
- other actions ( repair, audit an GE ) are also stored in storagenode_bandwdith_rollups and we can use them for statistics, etc.
- long term we should also fix bucket_bandwidth_rollups primary key but we can do this when load to this table will be decreased

**Comment by mniewrzal** [comment]
Changes were merged. We need to wait for release and deployment to see if that helps.

**Comment by mniewrzal** [comment]
Next attempt to fix this will be correcting primary key for `bucket_bandwidth_rollups` table from `bucket_name, project_id, interval_start, action` to `project_id, bucket_name, interval_start, action`. We will first apply new primary key to all satellites manually and meanwhile we will add migration code to reflect this change in source code. Query to execute will be 
```SQL
ALTER TABLE bucket_bandwidth_rollups ALTER PRIMARY KEY USING COLUMNS ( project_id, bucket_name, interval_start, action )
```

cc @thepaul

**Comment by thepaul** [comment]
If using `ALTER PRIMARY KEY`, the old primary key will be left intact as a `UNIQUE` constraint. I imagine we should drop that afterward as well

**Comment by mniewrzal** [comment]
> If using ALTER PRIMARY KEY, the old primary key will be left intact as a UNIQUE constraint. I imagine we should drop that afterward as well

Yes, that's true. We also need to cleanup code which is sorting entries before inserting into DB and ordering is using old columns order.

**Comment by mniewrzal** [comment]
For now situation calm down so lets move it for now to backlog.
